core/permission/access.go

```diff
@@ -51,6 +54,84 @@ func (a Access) Validate() error {
 	return errors.NotValidf("access level %s", a)
 }
 
+// ObjectType is the type of the permission object/
+type ObjectType int
+
+// These values must match the values in the permission_object_type table.
+const (
+	Cloud ObjectType = iota
```

Comment:
I did ask @manadart about this exact thing. He said iota is okay to use. I was copying from somewhere else actually, which isn't a great argument. It's a bit ugly here as one type is string, one is int. 

We should come to a consensus before we're unable to change the DB easily. 

It's easier to use itoa in the here as any of these validation failures is due to bad programing and not a user issue. But I am changing.

Reply:
Personally I don't like iota as a pattern. If it is only used for ephemeral in-memory only values, they can be fine. But as soon as they are persisted somewhere else, the chance that someone accidentally adds a const 'in the middle' of a block, because they think it makes contextual sense, you've now broken all the values after it. Having an explicit
 = 1, = 2, = 3 is slightly more work, but barely, and also means that if saw a "3" in the database, reversing the lookup is trivial with a search, rather than counting the number of entries.
 (even worse if you are using complex iotas like powers of two, etc.)